Assets/HoloToolkit/Input/Scripts/GestureManager.cs

```diff
@@ -60,183 +70,408 @@ public partial class GestureManager : Singleton<GestureManager>
 
-        private InteractionSourceState currentHandState;
+        private HashSet<uint> pressedInteractionSource = new HashSet<uint>();
 
-        void Start()
+        private bool hasRecognitionStarted;
+
+        private GameObject lastFocusedObject;
+
+        private void Awake()
         {
+            InteractionManager.SourceDetected += InteractionManager_SourceDetected;
             InteractionManager.SourcePressed += InteractionManager_SourcePressed;
             InteractionManager.SourceReleased += InteractionManager_SourceReleased;
             InteractionManager.SourceUpdated += InteractionManager_SourceUpdated;
             InteractionManager.SourceLost += InteractionManager_SourceLost;
 
             // Create a new GestureRecognizer. Sign up for tapped events.
+            // Will regester Taps for both Hand and Clicker.
             gestureRecognizer = new GestureRecognizer();
             gestureRecognizer.SetRecognizableGestures(GestureSettings.Tap);
-
-            manipulationRecognizer = new GestureRecognizer();
-            manipulationRecognizer.SetRecognizableGestures(GestureSettings.ManipulationTranslate);
-
             gestureRecognizer.TappedEvent += GestureRecognizer_TappedEvent;
 
             // We need to send pressed and released events to UI so they can provide visual feedback
             // of the current state of the UI based on user input.
             gestureRecognizer.RecognitionStartedEvent += GestureRecognizer_RecognitionStartedEvent;
             gestureRecognizer.RecognitionEndedEvent += GestureRecogniser_RecognitionEndedEvent;
 
+            // Start looking for gestures.
+            gestureRecognizer.StartCapturingGestures();
+
+            // Create a new GestureRecognizer. 
+            // Sign up for manipulation events.
+            manipulationRecognizer = new GestureRecognizer();
+
+            // ManipulationTranslate only recognizes hand gesture translations, but not Clicker translations.
+            manipulationRecognizer.SetRecognizableGestures(GestureSettings.ManipulationTranslate);
+
+            // Subscribe to our manipulation events.
             manipulationRecognizer.ManipulationStartedEvent += ManipulationRecognizer_ManipulationStartedEvent;
             manipulationRecognizer.ManipulationUpdatedEvent += ManipulationRecognizer_ManipulationUpdatedEvent;
             manipulationRecognizer.ManipulationCompletedEvent += ManipulationRecognizer_ManipulationCompletedEvent;
             manipulationRecognizer.ManipulationCanceledEvent += ManipulationRecognizer_ManipulationCanceledEvent;
 
-            // Start looking for gestures.
-            gestureRecognizer.StartCapturingGestures();
+            // Start looking for manipulation gestures.
             manipulationRecognizer.StartCapturingGestures();
         }
 
+        #region Interaction Management
+
+        /// <summary>
+        /// Thrown when we detect an interaction source.
```

Comment:
A bit of a nit, but I usually think that events are raised or set and exceptions are thrown.